tests/qos/test_buffer.py

```diff
@@ -0,0 +1,848 @@
+import logging
+import os
+import sys
+import time
+import re
+import json
+
+import pytest
+
+from tests.common import config_reload
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+
+profile_format = 'pg_lossless_{}_{}_profile'
+lossless_profile_pattern = 'pg_lossless_([1-9][0-9]*000)_([1-9][0-9]*m)_profile'
+
+default_cable_length_list = None
+default_lossless_headroom_data = None
+default_ingress_pool_number = 0
+default_mtu = '9100'
+
+testparam_headroom_override = None
+testparam_lossless_pg = None
+
+buffer_model_dynamic = True
+
+def detect_buffer_model(duthost):
+    global buffer_model_dynamic
+    buffer_model = duthost.shell('redis-cli -n 4 hget "DEVICE_METADATA|localhost" buffer_model')['stdout']
+    buffer_model_dynamic = buffer_model == 'dynamic'
+
+
+def detect_ingress_pool_number(duthost):
+    global default_ingress_pool_number
+    pools = duthost.shell('redis-cli -n 4 keys "BUFFER_POOL|ingress*"')['stdout']
+    default_ingress_pool_number = len(pools.split())
+
+
+def load_lossless_headroom_data(duthost):
+    global default_lossless_headroom_data
+    if not default_lossless_headroom_data:
+        dut_hwsku = duthost.facts["hwsku"]
+        dut_platform = duthost.facts["platform"]
+        skudir = "/usr/share/sonic/device/{}/{}/".format(dut_platform, dut_hwsku)
+        lines = duthost.shell('cat {}/pg_profile_lookup.ini'.format(skudir))["stdout"]
+        default_lossless_headroom_data = {}
+        for line in lines.split('\n'):
+            if line[0] == '#':
+                continue
+            tokens = line.split()
+            speed = tokens[0]
+            cable_length = tokens[1]
+            size = tokens[2]
+            xon = tokens[3]
+            xoff = tokens[4]
+            if not default_lossless_headroom_data.get(speed):
+                default_lossless_headroom_data[speed] = {}
+            default_lossless_headroom_data[speed][cable_length] = {'size': size, 'xon': xon, 'xoff': xoff}
+        default_lossless_headroom_data = default_lossless_headroom_data
+
+
+def load_test_parameters(duthost):
+    global default_cable_length_list
+    global testparam_headroom_override
+    global testparam_lossless_pg
+    with open(r"qos/files/dynamic_buffer_param.json") as file:
+        params = json.load(file)
+        logging.info("Got test parameters {}".format(params))
```

Comment:
Suggest to replace Got to Load. should you add the path of the file the parameters are loaded from?